taca/illumina/HiSeqX_Runs.py

```diff
@@ -255,28 +364,46 @@ def _generate_clean_samplesheet(ssparser, indexfile, fields_to_remove=None, rena
 
     return output
 
-def look_for_lanes_with_10X_indicies(indexfile, ssparser):
+
+def _classify_samples(indexfile, ssparser):
     """
     Given an ssparser object
-    returns a list of lanes with 10X indicies
+    goes through all samples and decide sample types
     """
-    index_dict=parse_10X_indexes(indexfile)
-    tenX_lanes = set() #Set to only get each lane once
-    not_tenX_lanes = set()
-    tenX_samples = dict()
+    sample_table = dict()
+    index_dict = parse_10X_indexes(indexfile)
     for sample in ssparser.data:
         lane = sample['Lane']
-        if sample['index'] in index_dict.keys():
-            tenX_lanes.add(lane)
-            sample_name = sample.get('Sample_Name') or sample.get('SampleName')
-            if tenX_samples.get(lane):
-                tenX_samples[lane].append(sample_name)
-            else:
-                tenX_samples.update({lane:[sample_name]})
+        sample_name = sample.get('Sample_Name') or sample.get('SampleName')
+        # 10X Genomic DNA & RNA
+        if TENX_GENO_PAT.findall(sample['index']):
+            index_length = [len(index_dict[sample['index']][0]),0]
+            sample_type = '10X_GENO'
+        # 10X scATAC
+        elif TENX_ATAC_PAT.findall(sample['index']):
+            index_length = [len(index_dict[sample['index']][0]),16]
+            sample_type = '10X_ATAC'
+        # UMI samples
+        elif UMI_IDX.findall(sample['index']) or UMI_IDX.findall(sample['index2']):
+            # Index length after removing "N" part
+            index_length = [len(sample['index'].replace('N','')),len(sample['index2'].replace('N',''))]
+            sample_type = 'UMI'
+        # No Index case. Note that if both index 1 and 2 are empty, it will be the same index type but will be handled in the next case
+        elif sample['index'].upper() == 'NOINDEX':
+            index_length = [0,0]
+            sample_type = 'ordinary'
+        # Ordinary samples
+        else:
+            index_length = [len(sample['index']),len(sample['index2'])]
+            sample_type = 'ordinary'
+
+        # Write in sample table
+        if sample_table.get(lane):
+            sample_table[lane].append({sample_name:{'sample_type':sample_type,'index_length':index_length}})
```

Comment:
I think we can replace all the `sample[sample.keys()[0]]` if you change each "row" of the sample_table to contain a tuple instead of a dictionary, like this: 
`sample_table[lane].append((sample_name, {'sample_type':sample_type,'index_length':index_length}))`

Reply:
I tested and the `tuple` solution didn't work. `sample_name` is in the same level as the following `dict`